agent/src/main/scala/za/co/absa/atum/agent/core/MeasurementProcessor.scala

```diff
@@ -0,0 +1,13 @@
+package za.co.absa.atum.agent.core
+
+import org.apache.spark.sql.DataFrame
+
+trait MeasurementProcessor {
+
+  type MeasurementFunction = DataFrame => String
```

Comment:
Would put it into the companion object. Not to face path-dependent type limitations